docs/adr/adr-007-public-api.md

```diff
@@ -0,0 +1,194 @@
+# ADR 007: Public API
+
+## Authors
+@Wondertan
+
+## Changelog
+
+- 2022-03-08: initial version
+
+## Context
+
+Celestia Node is being built for almost a half of a year with the bottom-up approach to development. The core lower 
+level components are built first and public API around them is getting organically shaped. The project has raised some 
+meat and the time to define the v0 API has come.
+
+## Alternative Approaches
+
+### Node Type centric Design
+Another discussed approach to defining an API could be an onion style, where each new layer is a feature set of bigger TODO Needs a better word
+node type. However, it turns out that this approach does not seem to match the reality we have, where each node type
+implements all the possible APIs but with the variable implementations matching resource constraints of a type.
+
+## Design
+
+### Goals
+* Ergonomic. Simple, idiomatic and self-explanatory.
+* Module centric(modular). The API is not monolithic and segregated into different categorized and independent modules.
+* Unified. All the node types implement the same set of APIs. The difference is defined by different implementations of
+some modules to meet resource requirements of a type. Example: FullAvailability and LightAvailability
+* Embeddable. Simply constructable Node with library style API. Not an SDK/Framework which dictates users the way to build
+an app, but users are those who decide how they want to build the app using the API.
+* Language agnostic. It should be simple to implement similar module interfaces/traits in other languages over RPC clients
+
+### Details
```

Comment:
```suggestion
### High level description

All node types in `celestia-node` will be referred to as `data availability nodes (DA nodes)` whose sole purpose is to interact with the `data availability network layer (DA layer)` such that the node contains all functionality necessary to post and retrieve messages from the DA layer.

This means that DA nodes will be able to query for / modify celestia state such that the DA nodes are able to pay for posting their messages on the network. The state-related API will be documented below in detail.

Furthermore, interaction between the celestia consensus network and the celestia data availability network will be the responsibility of the **bridge** DA node type. However, that interaction will not be exposed on a public level (meaning a **bridge** node will not expose the same RPC interface as the celestia-core node to which it is connected). A **bridge** node, for all intents and purposes, will provide the same API as that of a **full** node.

### Details
```

Reply:
cc @liamsi Please read through this and give your thoughts bc it's incredibly important we all align on what a DA node provides in terms of public API.

Reply:
The public API should also enable to retry DASing the data behind headers. Also querying the status of the celestia light node (height etc).

Reply:
> The public API should also enable to retry DASing the data behind headers. Also querying the status of the celestia light node (height etc).

This should be a suggestion on the ADR itself imo since it has more to do with method available

Reply:
> The public API should also enable to retry DASing the data behind headers.
`ShareModule.IsAvailable`